tests/test_conversationhandler.py

```diff
@@ -1325,112 +1334,139 @@ def slowbrew(_update, context):
         assert handler.conversations.get((self.group.id, user1.id)) is None
         assert self.is_timeout
 
-    def test_conversation_timeout_warning_only_shown_once(self, recwarn):
+    def test_handlers_generate_warning(self, recwarn):
+        """
+        this function tests all handler + per_* setting combinations.
+        """
+
+        # the warning message action needs to be set to always,
+        # otherwise only the first occurrence will be issued
+        filterwarnings(action="always", category=PTBUserWarning)
+
+        # this class doesn't do anything, its just not the Update class
+        class NotUpdate:
+            pass
+
+        # this conversation handler has the string, string_regex, Pollhandler and TypeHandler
+        # which should all generate a warning no matter the per_* setting. TypeHandler should
+        # not when the class is Update
         ConversationHandler(
-            entry_points=self.entry_points,
+            entry_points=[StringCommandHandler("code", self.code)],
             states={
-                self.THIRSTY: [
-                    ConversationHandler(
-                        entry_points=self.entry_points,
-                        states={
-                            self.BREWING: [CommandHandler('pourCoffee', self.drink)],
-                        },
-                        fallbacks=self.fallbacks,
-                    )
-                ],
-                self.DRINKING: [
-                    ConversationHandler(
-                        entry_points=self.entry_points,
-                        states={
-                            self.CODING: [CommandHandler('startCoding', self.code)],
-                        },
-                        fallbacks=self.fallbacks,
-                    )
+                self.BREWING: [
+                    StringRegexHandler("code", self.code),
+                    PollHandler(self.code),
+                    TypeHandler(NotUpdate, self.code),
                 ],
             },
-            fallbacks=self.fallbacks,
-            conversation_timeout=100,
-        )
-        assert len(recwarn) == 1
-        assert str(recwarn[0].message) == (
-            "Using `conversation_timeout` with nested conversations is currently not "
-            "supported. You can still try to use it, but it will likely behave "
-            "differently from what you expect."
+            fallbacks=[TypeHandler(Update, self.code)],
         )
-        assert recwarn[0].filename == __file__, "incorrect stacklevel!"
```

Comment:
why was this removed but not added back? If it was a mistake, then you can use a for loop to iterate over warnings and then check it.

Reply:
Im not sure what this assert checks, I can readd it.

Also not sure what you meant with for loop, do you want to go over all recwarnings and assert the filename?

Reply:
> Im not sure what this assert checks, I can readd it.

It checks if you passed in the correct `stacklevel` parameter to `warn()`. It does this by checking if the filename of the warning emitted matches the filename of the current module.

> Also not sure what you meant with for loop, do you want to go over all recwarnings and assert the filename?

yep exactly that